Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearch_details.cpp

```diff
@@ -180,8 +180,13 @@ std::vector<std::vector<std::unique_ptr<ROMol>>> splitMolecule(
   if (maxBondSplits < 1) {
     maxBondSplits = 1;
   }
-  maxBondSplits = std::min({maxBondSplits, MAX_CONNECTOR_NUM, query.getNumBonds()});
+  maxBondSplits =
+      std::min({maxBondSplits, MAX_CONNECTOR_NUM, query.getNumBonds()});
   const auto ringInfo = query.getRingInfo();
+  if (ringInfo->bondRings().empty()) {
```

Comment:
The direct way to check whether or not the ring finding has been done is:
```suggestion
  if (!ringInfo->isInitialized()) {
```
This has the additional advantage that you won't call `findSSSR` on every search when the query has no rings.